### PR TITLE
DOCS-289 Add Web3 Guide to Quickstarts

### PR DESCRIPTION
**Pull request by alexisintech**
The Web3 guide was outdated and had redundant information for setting up a Nextjs application.

This PR

- updates the Web3 guide to follow consistent patterns and language conventions, aligning it with the style used in the quickstart guides
- updates code examples to be functional

For reference, the old Web3 guide looked like this:

https://github.com/clerkinc/clerk-docs/assets/98043211/006dc25c-ec64-4e96-9096-cf5f5cfe9951

Fixes DOCS-289

Related to [this PR in marketing repo](https://github.com/clerkinc/clerk-marketing/pull/761)

### CODE REVIEW

File: docs/manifest.json [hunk 1]
```diff
@@ -104,7 +104,7 @@
       ["<UserProfile />", "/components/user/user-profile"],
       "# Guides",
       ["Sync data to your backend", "/users/sync-data"],
-      ["Web3 authentication", "/users/web3-wallets"]
```
Comment (BRKalow) [inline docs/manifest.json:107]: ❓ `Web3 authentication` feels more descriptive than just `Web3`. Why the change?

File: docs/users/web3.mdx
Comment (BRKalow) [inline docs/users/web3.mdx:1]: Let's make sure to add a redirect given that we're changing the file path

File: docs/users/web3.mdx [hunk 2]
```diff
@@ -0,0 +1,148 @@
...
+description: Learn to install and initialize Clerk in a new Web3 application.
+---
+
+# Use Clerk with Web3
+
+Learn how to use Clerk to quickly and easily add secure authentication and user management to your Web3 application.
+
+## Before you start
+
+You need to create a Clerk application in your [Clerk Dashboard](https://dashboard.clerk.com/) and configure the application to use **Sign in with Metamask**. For more information on how to set up a Clerk application, check out our [setup guide](/docs/quickstarts/setup-clerk). To enable Metamask authentication, check out [this guide](/docs/authentication/configuration/sign-in-options#social-o-auth-web3).
```
Comment (BRKalow) [inline docs/users/web3.mdx:12]:
```suggestion
You need to create a Clerk application in your [Clerk Dashboard](https://dashboard.clerk.com/) and configure the application to use **Sign in with Metamask**. For more information on how to set up a Clerk application, check out the [setup guide](/docs/quickstarts/setup-clerk). To enable Metamask authentication, check out [this guide](/docs/authentication/configuration/sign-in-options#social-o-auth-web3).
```

File: docs/users/web3.mdx
Comment (alexisintech) [reply docs/users/web3.mdx:1]: https://github.com/clerkinc/clerk-marketing/pull/761

File: docs/users/web3.mdx [hunk 3]
```diff
@@ -0,0 +1,148 @@
...
+```
+
+## Accessing the Web3 address from the backend
+
+### Add authentication to a serverless API route
+
+A new Next.js app comes with a sample API route in `/pages/api/hello.js`. Let's modify it to retrieve the user's Web3 address:
+
+```jsx filename="pages/api/hello.js" copy
+import { requireSession } from "@clerk/nextjs/api";
```
Comment (SokratisVidros) [inline docs/users/web3.mdx:110]: This API is a bit older. To avoid content duplication it would be ideal to link to https://clerk.com/docs/quickstarts/nextjs#api-routes

File: docs/users/web3.mdx [hunk 4]
```diff
@@ -0,0 +1,148 @@
...
+description: Learn to install and initialize Clerk in a new Web3 application.
+---
+
+# Use Clerk with Web3
+
+Learn how to use Clerk to quickly and easily add secure authentication and user management to your Web3 application.
+
+## Before you start
+
+You need to create a Clerk application in your [Clerk Dashboard](https://dashboard.clerk.com/) and configure the application to use **Sign in with Metamask**. For more information on how to set up a Clerk application, check out the [setup guide](/docs/quickstarts/setup-clerk). To enable Metamask authentication, check out [this guide](/docs/authentication/configuration/sign-in-options#social-o-auth-web3).
```
Comment (SokratisVidros) [inline docs/users/web3.mdx:12]: The anchor contains the phrase `social-o-auth`. It needs to be replaced with `web3-metamask` as this is not a social login connection like Google, or Facebook.

**Review by SokratisVidros — CHANGES_REQUESTED** [review]

File: docs/users/web3.mdx (hunk 4 above)
Comment (alexisintech) [reply docs/users/web3.mdx:12]: It's because it's linking to a section that talks about adding social oauth *plus* web3 auth. I can remove this link altogether and lay out the instructions (to adding web3 auth) here instead, if that feels more accurate/helpful.

File: docs/users/web3.mdx [hunk 5]
```diff
@@ -0,0 +1,119 @@
...
+
+## Creating a new Next.js app
+
+Once you have a Clerk application set up in the dashboard, it's time to create a Next.js application. Clerk provides a detailed guide on how to [create a new Next.js app with Clerk](/docs/quickstarts/nextjs).
+
+## Accessing the Web3 address from the frontend
+
+At this point, you should have a Next.js application integrated with Clerk and with Metamask authentication enabled. Run `npm run dev` and visit `localhost:3000`. If you followed the guide, all of the pages to your application should be protected, and you should see your sign-in page. 
+
+Your Clerk application will use the [Account Portal](/docs/account-portal/overview) pages out of the box. However, if you would like more control over your authentication flow, Clerk provides a set of [components](/docs/components/overview) that you can use. If the prebuilt components don't meet your specific needs, you can build fully [custom sign-up and sign-in flows](/docs/custom-flows/overview).
```
Comment (BRKalow) [inline docs/users/web3.mdx:22]: I feel like this paragraph can be removed, it seems out of place in the context of this content. What do you think?

**Review by BRKalow — APPROVED** [review]

**Review by SokratisVidros — APPROVED** [review]
